source/develop/release-management/features/storage/pass_discard_from_guest_to_underlying_storage.html.md

```diff
@@ -0,0 +1,234 @@
+At first sight, it seems that if *vm1* would have removed some files from *disk1* and discarded the unused space, then *lv2* could be extended and *vm2* could write to *disk2*. But that's not the case.
+
+The reason for this is the implementation of lvm. When *vm1* wrote 10G to *disk1* and filled *sd1* completely, we could see in lvm's metadata that *vg1* is full, so that a logical volume in *vg1* could not be given with additional logical extents anymore.<br/>
+After *vm1* discarded some unused blocks, lvm's metadata didn't change. That is, the size of *lv1* was still the same, and so is the size of *vg1* - it was still full. So when we tried to extend *lv2*, we couldn't, because there where no free logical extents in *vg1*.
+
+We said before that if we free some blocks from a disk that resides on a storage domain named *sd1*, other disks on *sd1* will not be able to use it if *sd1* went out of space.<br/>
+So now we can better understand who **can** consume those discarded blocks:
+
+1. Disks on other storage domains - because our storage is not full; and if it was full - we've just discarded a few blocks that can now be consumed.
+2. Other disks on *sd1* if *sd1* is not out of space - because if the vg is not full, we can extend its logical volumes with any blocks that are given to us by the storage, and those discarded blocks are not exceptional.
+3. In fact, if *sd1* went out of space, and we discarded enough blocks to extend it, we can always do that by adding another lun to the domain or by extending the existing one using the [Refresh LUN Size](/develop/release-management/features/storage/lun-resize), making it possible for our logical volumes to extend.
+
+#### File storage
+Unlike block storage, in file storage we don't have restrictions caused by metadata, so if the underlying file system supports discard, it will work in any scenario.<br/>
+If we take the example from above - after *vm2* was unable to write to *disk2*, if *vm1* removed some files and discarded the removed blocks from *disk1*, then:
+
+* The actual size of *disk1* reduces.
+* The amount of free space in *sd1* gets bigger.
+* *vm2* can write to *disk2*.
+
+### Restrictions
+
+#### The underlying storage
+
+##### Block storage
+The underlying storage must support *discard* calls in order for this to work.<br/>
+That's why when trying to enable *"Pass Discard"* for a VM disk, the engine first queries the disk's storage domain for its *discard* support.
+
+This is how we know if the storage supports *discard*:
+
+- Let *lunX* be a device and *dm-X* be its corresponding dm device for a natural number X. Then *lunX* is considered to ***support discard*** iff the value of */sys/block/dm-X/queue/discard_max_bytes* is bigger than 0 [^1].
+- A storage domain *supports discard* iff **all** of the luns that it is built from *support discard*. That means that if at least one of its luns doesn't *support discard*, then so is the storage domain.
+
```

Comment:
The discard support will be calculated for a storage domain, among other flows, after it is added with another lun.
So if that lun breaks the support, the storage domain's disks' *Pass Discard* field will be set to false.